metadata_layer.py

```diff
@@ -943,7 +943,7 @@ def from_edition(cls, edition):
         for contribution in edition.contributions:
             contributor = ContributorData.from_contribution(contribution)
             contributors.append(contributor)
-        else:
+        if not edition.contributions:
```

Comment:
Yes, that can still happen. This branch is only concerned with stopping a contributor from being listed as both PRIMARY_AUTHOR and AUTHOR; and if a contributor _does_ have multiple authorship roles in a book, stopping them from showing up twice in the list of contributors.